packages/cpp/src/browser/cpp-keybinding.ts

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright (C) 2017 Ericsson and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+import { injectable, inject } from "inversify";
+import { EditorManager } from "@theia/editor/lib/browser"
+import {
+    KeybindingContext, Keybinding, KeybindingContribution, KeybindingRegistry, KeyCode, Key, Modifier
+} from "@theia/core/lib/common";
+
+@injectable()
+export class CppKeybindingContext implements KeybindingContext {
+    constructor( @inject(EditorManager) protected readonly editorService: EditorManager) { }
+
+    id = 'cpp.keybinding.context';
+
+    isEnabled(arg?: Keybinding) {
+        return this.editorService && !!this.editorService.activeEditor &&
```

Comment:
- whether this.editorService is undefined check seems to be redundant
- extracting activeEditor to the const will make the code more readable 
- it seems that this code is duplicated in the PR

Reply:
Haven't looked at what causes the bug with the command palette yet.

Reply:
It should be fine with the latest master. @svenefftinge fixed it